batcher/aligned-batcher/src/lib.rs

```diff
@@ -1646,14 +1651,30 @@ impl Batcher {
         )
         .await
         {
-            Ok(_) => info!("createNewTask transaction successfully canceled"),
+            Ok(receipt) => {
+                info!("createNewTask transaction successfully canceled");
+                let gas_cost =
+                    Self::calculate_gas_cost(receipt.effective_gas_price, receipt.gas_used);
+                self.metrics
+                    .batcher_gas_cost_cancel_task_total
+                    .inc_by(gas_cost.as_u64());
+            }
             Err(e) => error!("Could not cancel createNewTask transaction: {e}"),
         };
         self.metrics
             .cancel_create_new_task_duration
             .set(start.elapsed().as_millis() as i64);
     }
 
+    fn calculate_gas_cost(gas_price: Option<U256>, gas_used: Option<U256>) -> U256 {
+        if let (Some(gas_price), Some(gas_used)) = (gas_price, gas_used) {
+            let wei_gas_cost = gas_price * gas_used;
+            let gwei_gas_cost = wei_gas_cost / U256::from(10e9 as i64);
```

Comment:
I think this conversion is not right, should be `1e9` instead

Reply:
+1